app/views/dashboard/firms/_form.html.erb

```diff
@@ -0,0 +1,59 @@
+<% if @firm.errors.any? %>
+  <div class="t-validation-summary">
+    <%= f.validation_summary %>
+  </div>
+<% end %>
+
+<section class="form-section">
+  <div class="form-section__heading-with-button">
+    <% if f.object.trading_name? %>
+      <h2 class="form-section__heading-with-button__heading"><%= t('dashboard.firm_edit.trading_name_heading') %></h2>
+    <% else %>
+      <h2 class="form-section__heading-with-button__heading"><%= t('dashboard.firm_edit.firm_heading') %></h2>
+      <a class="form-section__heading-with-button__button button button--primary"><%= t('dashboard.add_trading_name_button') %></a>
```

Comment:
I know this was just moved around on this PR but might've missed it before. This classlist got a little crazy.

It might be an idea to use a layout in this context. This would essentially say that all `button`'s in this layout should look have this style. For example:

```css
.l-form-section .button {
  float: right;
}
``` 

Its a cleaner way to override component's initial styles and removes the need to have a big long class list like this one. 

It also avoids components styling the same element and therefore competing for specificity (which I'm guessing is why https://github.com/moneyadviceservice/rad/commit/edf01a33130f37c8da45d394ee7acc91b2f7fdfe might've been there?). With a layout class, you're nearly always one level more 'specific'.

You could just wrap each `form-section` in a `l-form-section` container, or all of them in one – whichever you feel works best.